tests/POSStressTest.cpp

```diff
@@ -0,0 +1,87 @@
+#include <stdio.h>
+#include "verifier.hpp"
+
+#include <sstream>
+#include <string>
+#include <fstream>
+#include <thread>
+
+std::vector<uint8_t> HexToBytes(const char *hex_proof) {
+    int len = strlen(hex_proof);
+    assert(len % 2 == 0);
+    std::vector<uint8_t> result;
+    for (int i = 0; i < len; i += 2)
+    {
+        int hex1 = hex_proof[i] >= 'a' ? (hex_proof[i] - 'a' + 10) : (hex_proof[i] - '0');
+        int hex2 = hex_proof[i + 1] >= 'a' ? (hex_proof[i + 1] - 'a' + 10) : (hex_proof[i + 1] - '0');
+        result.push_back(hex1 * 16 + hex2);
+    }
+    return result;
+}
+
+void doit(int thread)
+{
+    std::ifstream infile("pos.txt");
```

Comment:
I think there's a risk this won't be as "stressing" is it looks, since every thread will sleep in disk I/O in every iteration of the loop. I would think it would put a lot more load (and stress) by loading and parsing the file once in the main thread, and have the worker threads just read from there